src/rasmus_fuel/realistic.py

```diff
@@ -0,0 +1,149 @@
+
+    u_wind_rel = wind_speed * cos(wind_dir_rel) - u_ship_og
+    v_wind_rel = wind_speed * sin(wind_dir_rel) - v_ship_og
+
+    coeff_wind_drag = 0.5* model_param["air_mass_density"]* model_param["wind_resistance_coefficient"] * model_param["vessel_supersurface_area"]
+
+    coeff_wave_drag = 20. * (model_param["waterline_width"] / model_param["waterline_length"]) ** (-1.2) * (1 / model_param["waterline_length"]) ** 0.62 \
+         / model_param["total_propulsive_efficiency"] / model_param["reference_froede_number"] * model_param["spectral_average"] * model_param["surface_water_density"] * \
+         model_param["waterline_width"] ** 2 * (model_param["acceleration_gravity"]/model_param["waterline_length"]**3)**0.05 * model_param["vessel_draught"] ** 0.62 * 0.25
+
+    power_needed = coeff * (speed_tw ** 3) + coeff_wind_drag * (u_wind_rel ** 2 + v_wind_rel ** 2) * speed_tw + coeff_wave_drag * w_wave_hight ** 2 * speed_tw
+
+    return power_needed
+
+
+def power_to_fuel_burning_rate(
+    power: npt.ArrayLike = None, efficiency: float = 0.5, fuel_value: float = 42.0e6
+) -> npt.ArrayLike:
+    """Convert power to fuel buring rate.
+    Parameters
+    ----------
+    power: array
+        Power in W (=kg*m2/s3).
+    efficiency: float
+        Fraction of fuel value turned into propulsive force. Defaults to 0.5.
+    fuel_value: float
+        Fuel value in J/kg. Defaults to 42.0e6.
+    Returns
+    -------
+    array
+        Fuel burning rate in kg/s.
+    """
+    fuel_burning_rate = power / efficiency / fuel_value
+    return fuel_burning_rate
+
+def power_to_fuel_consump(
+    engine_power: npt.ArrayLike = None,
+    steaming_time: npt.ArrayLike = None,
+    distance: npt.ArrayLike = None,
+) -> npt.ArrayLike:
+    """Convert engine power to fuel consumed per vessel weight per distance and unit time.
+    Parameters
+    ----------
+    engine_power: array
+        engine power in kWh .
+    steaming_time: array
+        sailing time [hours] for vessel
+    distance: array
+        sailing distance in m
+    Returns
+    -------
+    array
+        Fuel consumption kW/kg/m.
+    """
+    fuel_consump = model_param["specific_fuel_consumption"] * engine_power * steaming_time/ model_param["DWT"] / distance
+    return fuel_consump
+
+def energy_efficiency_per_time_distance(
+    fuel_consumption: npt.ArrayLike = None,
+)-> npt.ArrayLike:
+    """Convert engine power to fuel consumed per vessel weight per distance and unit time.
+    Parameters
+    ----------
+    fuel_consumption: array
+        fuel consumption kWh/kg/m
+    Returns
+    -------
+    array
+        energy efficiency indicator
+    """
+    energy_efficiency = model_param["conversion_factor_fuelmass2CO2"] * fuel_consumption
+    return energy_efficiency
+
+#def func_cost_function(ds_traj):
+    # Cost function is equal to engine fuel consumption in this case
+#    cost = ds_traj.fuel_consumption.sum("time_steps")
+#    return cost
```

Comment:
Remove?

Reply:
Handling the trajectory data sets is out of scope for this package. Here, we should focus on calculating the fuel consumption for given conditions.

Then, the exact form of the cost function and how it applies to a trajectory should be implemented closer to the code that does the analysis or optimisation.